core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -626,9 +626,18 @@ private int getGlobalTimeout() {
    }
 
    void broadcastTopologyUpdate(String cacheName, CacheTopology cacheTopology, AvailabilityMode availabilityMode) {
+      broadcastTopologyUpdate(cacheName, cacheTopology, availabilityMode, false);
+   }
+
+   void broadcastTopologyUpdate(String cacheName, CacheTopology cacheTopology, AvailabilityMode availabilityMode, boolean sync) {
       ReplicableCommand command = new TopologyUpdateCommand(cacheName, transport.getAddress(), cacheTopology,
             availabilityMode, viewId);
-      helper.executeOnClusterAsync(transport, command);
+      if (sync) {
+         CompletionStages.join(helper.executeOnClusterSync(
+               transport, command, getGlobalTimeout(), VoidResponseCollector.ignoreLeavers()));
+      } else {
```

Comment:
-1000 Blocking threads in the server is a bad thing due to the small thread pool size and most of our code is non-blocking.

Reply:
Did some more changes so we can return the completion stage instead.